src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -666,6 +675,40 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
            !$omp end parallel
          end if
 
+         ! When config_GM_closure = 'edenGreatbatch' actually modify the value of gmBolusKappa based on
+         ! Eden and Greatbach (2008), Ocean Model., but assuming quasi-geostrophy to simplify the EKE budget
+         if (local_config_GM_compute_edenGreatbatch) then
+           call mpas_pool_get_array(diagnosticsPool, 'betaEdge', betaEdge)
+           call mpas_pool_get_array(meshPool, 'fEdge', fEdge)
+           call mpas_pool_get_array(diagnosticsPool, 'RiTopOfCell', RiTopOfCell)
+
+           !$omp parallel
+           !$omp do schedule(runtime) private(cell1, cell2, sumRi, sigma, RiTopOfEdge, &
+           !$omp             L_rhines, Lr, Length)
+           do iEdge=1,nEdges
+              sumRi = 0.0_RKIND
+              cell1 = cellsOnEdge(1, iEdge)
+              cell2 = cellsOnEdge(2, iEdge)
+
+              do k=1,maxLevelEdgeTop(iEdge)
+                 RiTopOfEdge = 0.5_RKIND*(max(RiTopOfCell(k,cell1),0.0_RKIND) + &
+                                          max(RiTopOfCell(k,cell2),0.0_RKIND))
+                 sigma = max(abs(fEdge(iEdge)),sqrt(2.0*betaEdge(iEdge)*cGMphaseSpeed(iEdge))) / &
+                         sqrt(sumRi + config_gm_EG_riMin)
+                 L_rhines = sigma / (1.0E-11_RKIND + betaEdge(iEdge))
+                 Lr = min(cGMphaseSpeed(iEdge) / abs(fEdge(iEdge)), sqrt(cGMphaseSpeed(iEdge) / &
+                         (2.0_RKIND*betaEdge(iEdge))))
+                 Length = min(Lr,L_rhines)
+                 !For compatibility with other schemes, we normalize by the 2-D kappa field, which will
+                 !be removed below in the computation of the stream function
+                 gmKappaScaling(k,iEdge) = max(config_GM_visbeck_min_kappa, min(config_gm_EG_kappa_factor* &
+                         Length**2.0*sigma, config_GM_visbeck_max_kappa)) / gmBolusKappa(iEdge)
```

Comment:
@vanroekel this line had gmBolusKappa(iCell) rather than iEdge. I just fixed it. I think this could alter your E3SM cases with Eden & Greatbatch that you already ran.

BTW, if you are working in E3SM, you can do a quick test with _D for debug, like
```
./create_test SMS_Ln9_D.T62_oQU120.GMPAS-IAF
```
and it will catch those.

Reply:
thanks for the catch and the tip @mark-petersen!